cloudspanner/src/main/java/com/yahoo/ycsb/db/CloudSpannerClient.java

```diff
@@ -0,0 +1,399 @@
+    if (spanner != null) {
+      return spanner;
+    }
+    String numChannels = properties.getProperty(CloudSpannerProperties.NUM_CHANNELS);
+    int numThreads = Integer.parseInt(properties.getProperty(Client.THREAD_COUNT_PROPERTY, "1"));
+    SpannerOptions.Builder optionsBuilder = SpannerOptions.newBuilder()
+        .setSessionPoolOption(SessionPoolOptions.newBuilder()
+            .setMinSessions(numThreads)
+            // Since we have no read-write transactions, we can set the write session fraction to 0.
+            .setWriteSessionsFraction(0)
+            .build());
+    if (host != null) {
+      optionsBuilder.setHost(host);
+    }
+    if (project != null) {
+      optionsBuilder.setProjectId(project);
+    }
+    if (numChannels != null) {
+      optionsBuilder.setNumChannels(Integer.parseInt(numChannels));
+    }
+    spanner = optionsBuilder.build().getService();
+    Runtime.getRuntime().addShutdownHook(new Thread("spannerShutdown") {
+        @Override
+        public void run() {
+          System.out.println("Closing Cloud Spanner client...");
+          spanner.closeAsync();
+          System.out.println("...done.");
+        }
+      });
+    return spanner;
+  }
+
+  @Override
+  public void init() throws DBException {
+    synchronized (CLASS_LOCK) {
+      if (dbClient != null) {
+        return;
+      }
+      Properties properties = getProperties();
+      String host = properties.getProperty(CloudSpannerProperties.HOST);
+      String project = properties.getProperty(CloudSpannerProperties.PROJECT);
+      String instance = properties.getProperty(CloudSpannerProperties.INSTANCE, "ycsb-instance");
+      String database = properties.getProperty(CloudSpannerProperties.DATABASE, "ycsb-database");
+
+      fieldCount = Integer.parseInt(properties.getProperty(
+          CoreWorkload.FIELD_COUNT_PROPERTY, CoreWorkload.FIELD_COUNT_PROPERTY_DEFAULT));
+      queriesForReads = properties.getProperty(CloudSpannerProperties.READ_MODE, "query").equals("query");
+      batchInserts = Integer.parseInt(properties.getProperty(CloudSpannerProperties.BATCH_INSERTS, "1"));
+      constructStandardQueriesAndFields(properties);
+
+      int boundedStalenessSeconds = Integer.parseInt(properties.getProperty(
+          CloudSpannerProperties.BOUNDED_STALENESS, "0"));
+      timestampBound = (boundedStalenessSeconds <= 0) ?
+          TimestampBound.strong() : TimestampBound.ofMaxStaleness(boundedStalenessSeconds, TimeUnit.SECONDS);
+
+      try {
+        spanner = getSpanner(properties, host, project);
+        if (project == null) {
+          project = spanner.getOptions().getProjectId();
+        }
+        dbClient = spanner.getDatabaseClient(DatabaseId.of(project, instance, database));
+      } catch (Exception e) {
+        LOGGER.log(Level.SEVERE, "init()", e);
+        throw new DBException(e);
+      }
+
+      LOGGER.log(Level.INFO, new StringBuilder()
```

Comment:
I double checked that all logging goes to stderr by default. So unless people explicitly provide logging options to make these go to stdout, it should be fine.